### DISCUSSION TITLE
Using different static site generators

### DISCUSSION THREAD

**Discussion by AlMcKinlay** [community/community#21563] [Other Feature Feedback, Questions, & Ideas]
<p>I know GitHub supports Jekyll out of the box, and you can keep your source in the repository and it will automatically be compiled and deployed, but I was wondering if there was any support for other static site generators. I have recently started using <a href="https://gohugo.io/" rel="nofollow noopener">hugo</a> which I’m really enjoying using.</p>

**Comment by birdcar** [comment, marked as answer]
<p>GitHub Pages doesn’t support other static site generators to the same degree that we support Jekyll, but it’s totally possible to to use other static site generators to publish a GitHub Pages site. Additionally, you’ll often find that the documentation for your static site generator of choice contains a guide for how to publish your site to GitHub Pages. An example from Hugo’s documentation can be found here:</p>
<p><a href="https://gohugo.io/hosting-and-deployment/hosting-on-github/" rel="nofollow noopener">https://gohugo.io/hosting-and-deployment/hosting-on-github/</a></p>
<p>Every static site generator will have a slightly different way of doing things, but in general, you’ll need to do the following:</p>
<ol>
<li><a href="https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/#publishing-your-github-pages-site-from-a-docs-folder-on-your-master-branch" rel="nofollow noopener">Configure your repository to publish your site from a “/docs” folder on the “master” branch.</a></li>
<li>Configure your static site generator to build your site’s files in the “/docs” directory.</li>
<li>Test and build your site locally.</li>
<li>Commit your changes.</li>
<li>Push those changes to GitHub.</li>
</ol>
<p>And that’s it! GitHub Pages will now take the site that your static site generator has built in the “/docs” directory and will publish it as a GitHub Pages site.</p>

**Comment by nanisadw3** [comment]
Yes, absolutely! While Jekyll is natively supported out of the box, GitHub Pages now natively supports **any static site generator** (Hugo, Astro, Next.js, Vite, Eleventy, Nuxt, etc.) through **GitHub Actions**.
    
    ### How to use any generator:
    1. Go to your repository **Settings** -> **Pages**.
    2. Under **Build and deployment** -> **Source**, select **GitHub Actions** instead of "Deploy from a branch".
    3. GitHub will suggest ready-to-use workflow templates for Hugo, Next.js, Astro, Gatsby, etc., or you can
  write a simple `.github/workflows/deploy.yml` that runs your build script (e.g. `npm run build`) and uses
  `actions/deploy-pages@v4`.
    
    This completely decouples you from Jekyll and allows using modern tooling with zero hosting cost.
    
    If this answers your question, please mark it as accepted!
